docs/ruleset_functions/get_building_segment_SWH_bat.md

```diff
@@ -0,0 +1,58 @@
+## get_building_segment_SWH_bat
+
+Description: This function determines the SWH BAT for the given building segment.
+
+Inputs:
+- **RMD**
+- **building_segment**
+
+Returns:
+- **building_segment_swh_bat**: one of the ServiceWaterHeatingSpaceOptions2019ASHRAE901 options
+
+Function Call:
+
+- get_energy_required_to_heat_swh_use  
+- get_SWH_uses_associated_with_each_building_segment  
+- get_SWH_uses_associated_with_each_building_segment  
+- get_component_by_ID  
+
+Data Lookup: None
+
+Logic:
+
+- set the building_segment_swh_bat to nil: `building_segment_swh_bat = "UNDETERMINED"`
+- if the building_segment.service_water_heating_building_area_type exists set the swh_bat to the one given: `if building_segment.service_water_heating_building_area_type:`
+    - set the building_segment_swh_bat to the building_segment.service_water_heating_building_area_type: `building_segment_swh_bat = building_segment.service_water_heating_building_area_type`
+- otherwise, if the building segment doesnt have service_water_heating_building_area_type, we need to determine the building segment SWH type by looking at individual SWH uses: `else:`
+    - create a dictionary that will hold the different types of swh_use_bat_types and the total service water used for the year: `swh_use_dict = {}`
+    - get the service water heating uses in the building segment `service_water_heating_use_ids = get_SWH_uses_associated_with_each_building_segment(RMD, building_segment.id)`
+    - look at each service water heating use id: `for swh_use_id in service_water_heating_use_ids:`
+        - get the swh_use using get_component_by_ID: `swh_use = get_component_by_ID(RMD, swh_use_id)`
+        - if any swh_use has use_units equal to "OTHER", the total energy required to heat the use cannot be determined, and we return "UNDETERMINED": `if swh_use.use_units == "OTHER": return "UNDETERMINED"`
+        - calculate the total hot water used using the function get_energy_required_to_heat_swh_use and add it to the swh_use_dict: `swh_use_dict[swh_use.area_type] += get_energy_required_to_heat_swh_use(swh_use, RMD)`
+        - check to see if the swh_use has service_water_heating_area_type: `if swh_use.area_type:`
+            - add the SWH building area type to the swh_use_dict and set the value to 0: `swh_use_dict.set_default(swh_use.area_type, 0)`
```

Comment:
If none of the swh_use BAT types are defined for a space, do we want to check to see if the space.service_water_heating_area_type was defined and use that in that specific case?

Reply:
I was not checking **space.service_water_heating_area_type** because that doesn't make a lot of sense to me if we are moving towards assigning service water heating uses at the building segment level instead of at the space level, though it certainly could be done.  Thoughts?

Reply:
If it wouldn't be much additional effort, just to have anther way to establish the BAT for the segment, I think it could be a good third option.

Reply:
ok, I did this in the simplest way possible while still maintaining encapsulation - I made a change to `get_energy_required_to_heat_swh_use`, so that it now returns a dictionary with space_ids as keys and energy used for each space as values.  For most uses, getting the total energy can be done by simply summing the values.  But for this function it allows us to look at the space BAT if needed.